### PR TITLE
Added method to test namespace to save JSON object to a file.

### PR DESCRIPTION
**Pull request by NissimHadar**
# Description
A method has been added to the Test namespace, available when running from the command line with the `--testScript` parameter.  This method saves an object to a file, in JSON format.  A new parameter `testResultsLocation` defines where the file will be saved.

The new command is Test.saveObject(<_object name_>, <_file name_>).

For example:
```
        var results = {
            elapsedTime: 0.05
        }
        
        Test.saveObject(results, "results.txt");

```

# Testing
The test will run a performance test in the dev-avatarislanfd domain.
1.  Create a folder to store the results in.  Assume it is C:\testing.
2.  Run the following line in a consoles:
```
interface.exe --testScript https://github.com/highfidelity/hifi_tests/blob/master/tests/performance/testAuto.js?raw=true quitWhenFinished --testResultsLocation C:\testing
```

3.  Verify that a file named results.txt has been created in C:\testing.  It's contents should be similar to the following:
```
{
    "elapsedTime": 11.544
}
```

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9699/HighFidelity-Beta-PR13154-39e5b578f8566a30ac07b795ccd9a56ce7203ed3.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9699/HighFidelity-Beta-Interface-PR13154-39e5b578f8566a30ac07b795ccd9a56ce7203ed3.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9699/HighFidelity-Beta-PR13154-39e5b578f8566a30ac07b795ccd9a56ce7203ed3.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9699/HighFidelity-Beta-Interface-PR13154-39e5b578f8566a30ac07b795ccd9a56ce7203ed3.exe

**Comment by samcake** [comment]
first performance has been merged in hifi_tests,
Available here:
https://github.com/highfidelity/hifi_tests/blob/master/tests/performance/test.js

File: interface/src/Application.h [hunk 1]
```diff
@@ -418,6 +418,7 @@ public slots:
     void updateVerboseLogging();
     Q_INVOKABLE void openAndroidActivity(const QString& activityName);
 
+    QString getTestResultsLocation() { return testResultsLocation; };
```
Comment (samcake) [inline interface/src/Application.h]: Why not part of the test Scripting interface ?

File: interface/src/Application.h [hunk 2]
```diff
@@ -751,7 +752,7 @@ private slots:
     std::atomic<bool> _pendingIdleEvent { true };
     std::atomic<bool> _pendingRenderEvent { true };
 
-    QString testSnapshotLocation;
+    QString testResultsLocation;
```
Comment (samcake) [inline interface/src/Application.h]: WHy not store this test specific information in the actual Test scripting interface class instead of in APplication ?

**Comment by samcake** [comment]
this is Failing Android, do you know why ?

**Comment by SamGondelman** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9781/HighFidelity-Beta-PR13154-d4d2f3c9573f375a8af51886a8e468a0c9500386.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9781/HighFidelity-Beta-PR13154-d4d2f3c9573f375a8af51886a8e468a0c9500386.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9781/HighFidelity-Beta-Interface-PR13154-d4d2f3c9573f375a8af51886a8e468a0c9500386.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9781/HighFidelity-Beta-PR13154-d4d2f3c9573f375a8af51886a8e468a0c9500386.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9781/HighFidelity-Beta-Interface-PR13154-d4d2f3c9573f375a8af51886a8e468a0c9500386.exe

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
File was created successfully.

**Comment by NissimHadar** [comment]
Who puts the *QA Approved* label?

**Pull request merged**